Src/CSharpier.Rider/src/main/java/com/intellij/csharpier/CustomPathInstaller.java

```diff
@@ -0,0 +1,37 @@
+package com.intellij.csharpier;
+
+import com.intellij.openapi.diagnostic.Logger;
+
+import java.io.File;
+import java.nio.file.Path;
+
+public class CustomPathInstaller {
+    Logger logger = CSharpierLogger.getInstance();
+
+    public void ensureVersionInstalled(String version) {
+        if (version == null || version.equals("")) {
+            return;
+        }
+        String directoryForVersion = getDirectoryForVersion(version);
+        File file = new File(directoryForVersion);
+        if (file.exists()) {
+            this.logger.debug("File at " + directoryForVersion + " already exists");
+            return;
+        }
+
+        String[] command = {"dotnet", "tool", "install", "csharpier", "--version", version, "--tool-path", directoryForVersion};
+        ProcessHelper.ExecuteCommand(command, null, null);
+    }
+
+    private String getDirectoryForVersion(String version) {
+        String userHome = System.getProperty("user.home");
+        Path path = Path.of(userHome, ".csharpier", version);
```

Comment:
Can we put it in `~/.cache/csharpier` instead of contributing to the littering of people's home directories?
https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

Reply:
+1, though on Windows you'd probably want a `System.getenv("LOCALAPPDATA")` prefix.

Reply:
Both good ideas, done.